pkg/querier/queryrange/split_by_interval.go

```diff
@@ -122,7 +122,13 @@ func (h *splitByInterval) Process(
 				return nil, data.err
 			}
 
-			responses = append(responses, data.resp)
+			if casted, ok := data.resp.(*LokiResponse); ok {
```

Comment:
If all the responses are empty, we would need to return a single empty response so Loki could distinguish between an error and an empty result set when merging them.

Reply:
Since both this and the existing if condition below are attempting to type assert and using the `ok` we could just do the type assertion once and use it in both places.